src/core/Akka.Remote/Transport/AkkaPduCodec.cs

```diff
@@ -477,14 +485,28 @@ private AcknowledgementInfo AckBuilder(Ack ack)
         /// <param name="seqOption">TBD</param>
         /// <param name="ackOption">TBD</param>
         /// <returns>TBD</returns>
-        public override ByteString ConstructMessage(Address localAddress, IActorRef recipient, SerializedMessage serializedMessage,
+        public override ByteString ConstructMessage(Address localAddress, IActorRef recipient,
+            SerializedMessage serializedMessage,
             IActorRef senderOption = null, SeqNo seqOption = null, Ack ackOption = null)
         {
             var ackAndEnvelope = new AckAndEnvelopeContainer();
-            var envelope = new RemoteEnvelope() { Recipient = SerializeActorRef(recipient.Path.Address, recipient) };
-            if (senderOption != null && senderOption.Path != null) { envelope.Sender = SerializeActorRef(localAddress, senderOption); }
-            if (seqOption != null) { envelope.Seq = (ulong)seqOption.RawValue; } else envelope.Seq = SeqUndefined;
-            if (ackOption != null) { ackAndEnvelope.Ack = AckBuilder(ackOption); }
+            var envelope = new RemoteEnvelope() { Recipient = _cache.Cache.GetOrCompute(recipient) };
+            if (senderOption != null && senderOption.Path != null)
+            {
+                envelope.Sender = _cache.Cache.GetOrCompute(senderOption);
```

Comment:
@Aaronontheweb, There doesn't happen to be a way to potentially re-use this on the receive side, is there? <_<

Reply:
Yep, different cache but it had a massive performance impact: https://github.com/akkadotnet/akka.net/pull/5240

About 40k msg/s

Reply:
@Aaronontheweb , I mean, I'm wondering if there would be any benefit to taking this serialized `ActorRefData` and the `IActorRef`, and putting them somewhere so that the `Receive` side would be able to use them on Decode, helping to avoid the first-resolve penalty on the caches in #5240 

Also, said impl could potentially be useful for `Ask` actors, since we -know- that we want a response on them. Not sure if I'm making sense.